tests/test_app/parser.h

```diff
@@ -1,26 +1,32 @@
 #ifndef VIRTUALIZE_META_LIBRARY_H
 #define VIRTUALIZE_META_LIBRARY_H
 
-#define MAX_NUM_IO 32
+#include "common/tiny-json.h"
+
+#define MAX_NUM_IO 4
+#define MAX_NUM_IO_TILES 8
 #define BUFFER_SIZE 1024
-#define MAX_NUM_PARSER 16
-#define GET_PLACE_INFO(info) struct PlaceInfo *place_info = \
-                                    (struct PlaceInfo *) info
+#define MAX_NUM_KERNEL 16
+#define MAX_JSON_FIELDS 512
+#define MAX_CONFIG 20
+#define MAX_ADDR_GEN_LOOP 5
 
 #define GET_KERNEL_INFO(info) struct KernelInfo *kernel_info = \
                                     (struct KernelInfo *) info
 
+#define GET_IO_INFO(info) struct IOInfo *io_info = \
```

Comment:
Does this macro have to be in the header file?

Reply:
Okay. I moved to src file.